examples/tutorials/nb_python/managed_rigid_object_tutorial.py

```diff
@@ -0,0 +1,560 @@
+
+    # video rendering
+    if make_video:
+        vut.make_video(
+            observations,
+            "rgba_camera_1stperson",
+            "color",
+            output_path + "local_velocity_control",
+            open_vid=show_video,
+        )
+
+    # [/local_velocity_control]
+    rigid_obj_mgr.remove_all_objects()
+    # %%
+    # [embodied_agent]
+
+    # load the lobot_merged asset
+    locobot_template_id = obj_templates_mgr.load_configs(
+        str(os.path.join(data_path, "objects/locobot_merged"))
+    )[0]
+
+    # add robot object to the scene with the agent/camera SceneNode attached
+    locobot = rigid_obj_mgr.add_object_by_id(
+        locobot_template_id, sim.agents[0].scene_node
+    )
+    locobot.translation = np.array([1.75, -1.02, 0.4])
+
+    vel_control = locobot.velocity_control
+    vel_control.linear_velocity = np.array([0, 0, -1.0])
+    vel_control.angular_velocity = np.array([0.0, 2.0, 0])
+
+    # simulate robot dropping into place
+    observations = simulate(sim, dt=1.5, get_frames=make_video)
+
+    vel_control.controlling_lin_vel = True
+    vel_control.controlling_ang_vel = True
+    vel_control.lin_vel_is_local = True
+    vel_control.ang_vel_is_local = True
+
+    # simulate forward and turn
+    observations += simulate(sim, dt=1.0, get_frames=make_video)
+
+    vel_control.controlling_lin_vel = False
+    vel_control.angular_velocity = np.array([0.0, 1.0, 0])
+
+    # simulate turn only
+    observations += simulate(sim, dt=1.5, get_frames=make_video)
+
+    vel_control.angular_velocity = np.array([0.0, 0.0, 0])
+    vel_control.controlling_lin_vel = True
+    vel_control.controlling_ang_vel = True
+
+    # simulate forward only with damped angular velocity (reset angular velocity to 0 after each step)
+    observations += simulate(sim, dt=1.0, get_frames=make_video)
+
+    vel_control.angular_velocity = np.array([0.0, -1.25, 0])
+
+    # simulate forward and turn
+    observations += simulate(sim, dt=2.0, get_frames=make_video)
+
+    vel_control.controlling_ang_vel = False
+    vel_control.controlling_lin_vel = False
+
+    # simulate settling
+    observations += simulate(sim, dt=3.0, get_frames=make_video)
+
+    # remove the agent's body while preserving the SceneNode
+    rigid_obj_mgr.remove_object_by_ID(locobot.ID, delete_object_node=False)
+
+    # video rendering with embedded 1st person view
+    if make_video:
+        vut.make_video(
+            observations,
+            "rgba_camera_1stperson",
+            "color",
+            output_path + "robot_control",
+            open_vid=show_video,
+        )
+
+    # [/embodied_agent]
+
+    # %%
+    # [embodied_agent_navmesh]
+
+    # load the lobot_merged asset
+    locobot_template_id = obj_templates_mgr.load_configs(
+        str(os.path.join(data_path, "objects/locobot_merged"))
+    )[0]
+    # add robot object to the scene with the agent/camera SceneNode attached
+    locobot = rigid_obj_mgr.add_object_by_id(
```

Comment:
Hmmm I see why it works that way.  This way you can easily attach a rigid object to the agent's scene node and then control it and its sensors will follow.

Reply:
How hard would it be to track internally if the object wrapper owns the scene node?  If `add_object_by_id` was given a scene node it is assumed that the object wrapper doesn't own that node (and thus doesn't delete it by default).  Then `delete_object_node` has a default value of `None` where that gets inferred as `True` if the wrapper is an owner and `False` if it isn't.  The user can override this inference by setting the value explicitly.

@mosra IIRC `Corrade::Containers::Optional` doesn't have python bindings right?  pybind11 treats pointers as optionals so a `bool*` would work as the signature that get's bound to python and we can bind a lambda so that the cpp API doesn't have to be ugly :)

Reply:
It has them since https://github.com/mosra/magnum-bindings/commit/8e00226da3b01364064d29ba64b594ea567d79b7 (@Skylion007 requested this), but I think it's not pulled in by Habitat yet.

Reply:
> How hard would it be to track internally if the object wrapper owns the scene node? If `add_object_by_id` was given a scene node it is assumed that the object wrapper doesn't own that node (and thus doesn't delete it by default). Then `delete_object_node` has a default value of `None` where that gets inferred as `True` if the wrapper is an owner and `False` if it isn't. The user can override this inference by setting the value explicitly.
> 

Right now the wrapper owns nothing but a weak ref to the object itself.  The manager just passes the arguments for add and remove into the physics manager.

Reply:
Looks like I got terminology wrong here.  If you swap "object wrapper" for "object" does that make sense?